dask/dataframe/core.py

```diff
@@ -2877,7 +2877,9 @@ def values(self):
         Operations that depend on shape information, like slicing or reshaping,
         will not work.
         """
-        return self.map_partitions(methods.values)
+        a = self.map_partitions(methods.values)
+        a._meta = methods.values(self._meta)
```

Comment:
It seems like this should already be happening in `_emulate` so I am wondering where that codepath goes wrong. https://github.com/dask/dask/blob/2613608da99b6a46333c0e2ff2000cf20d41faa5/dask/dataframe/core.py#L5558-L5561

Reply:
Yeah - `_emulate` actually works correctly.  I discovered that [this change](https://github.com/dask/dask/pull/7486#discussion_r604326049) fixes the issue for me.